modules/services/window-managers/bspwm/options.nix

```diff
@@ -0,0 +1,220 @@
+{ pkgs, lib }:
+
+with lib;
+
+let
+  rule = types.submodule {
+    options = {
+      monitor = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The monitor where the rule should be applied";
+        example = "HDMI-0";
+      };
+
+      desktop = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The desktop where the rule should be applied";
+        example = "^8";
+      };
+
+      node = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The node where the rule should be applied";
+        example = "1";
+      };
+
+      state = mkOption {
+        type = types.nullOr (types.enum [ "tiled" "pseudo_tiled" "floating" "fullscreen" ]);
+        default = null;
+        description = "The state in where the window should be spawned";
+        example = "floating";
+      };
+
+      layer = mkOption {
+        type = types.nullOr (types.enum [ "below" "normal" "above" ]);
+        default = null;
+        description = "The layer where the window should be spawned";
+        example = "above";
+      };
+
+      splitDir = mkOption {
+        type = types.nullOr (types.enum [ "north" "west" "south" "east" ]);
+        default = null;
+        description = "The direction where the container is going to be splitted";
+        example = "south";
+      };
+
+      splitRatio = mkOption {
+        type = types.nullOr types.float;
+        default = null;
+        description = "The ratio between the new window and the previous existing window in the desktop";
+        example = 0.65;
+      };
+
+      hidden = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node isn't going to occupy any space";
```

Comment:
The typical way to express would be `Whether …` instead of `If it's set to …`

For example, "Where the node should occupy any space."

Reply:
The description strings are taken directly from the bspwm manpages. Do you think it's worth rephrasing them?

Reply:
Yeah, I think so. I'm more concerned about maintaining consistent descriptions across HM modules than between the module and the upstream project.

Reply:
You got it.